app/src/main/java/ch/epfl/sweng/erpa/CreateGame.java

```diff
@@ -1,13 +1,141 @@
+                String feedbackType = universesSpinner.getSelectedItem().toString();
+                if(feedbackType.equals("Other"))
+                    universeField.setVisibility(View.VISIBLE);
+                else
+                    universeField.setVisibility(View.GONE);
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> parentView) {}
+        });
+    }
+
+    @Override
+    public void onFragmentInteraction(Uri uri) {
+        Log.i("coucou",uri.toString());
+    }
+
+    // When Oneshot or Campaign checkboxes are checked, uncheck the other one
+    public void onOneShotOrCampaignSelected(View view) {
+        ButterKnife.bind(this, view);
+        if (view.getId() == R.id.campaign) {
+            findViewById(R.id.num_sessions).setVisibility(View.VISIBLE);
+            findViewById(R.id.numb_session_field).setVisibility(View.VISIBLE);
+
+        } else {
+            findViewById(R.id.num_sessions).setVisibility(View.GONE);
+            findViewById(R.id.numb_session_field).setVisibility(View.GONE);
+        }
+    }
+
+    //call when the user submit a game and check if no requested field is empty
+    public void submitGame(View view) {
+        Spinner difficultySpinner = findViewById(R.id.difficulty_spinner);
+        Spinner universesSpinner = findViewById(R.id.universes_spinner);
+        Spinner sessionLengthSpinner = findViewById(R.id.session_length_spinner);
+        EditText minPlayer = findViewById(R.id.min_num_player_field);
+        EditText maxPlayer = findViewById(R.id.max_num_player_field);
+        String valueMin = minPlayer.getText().toString();
+        String valueMax = maxPlayer.getText().toString();
+        if(!playerNumberIsValid(valueMin, valueMax)){
+            createPopup(getString(R.string.invalidPlayerNumber));
+        }
+        else if(!allObligFieldsFilled()) {
+            createPopup(getString(R.string.emptyFieldMessage));
+        }
+        else if(!aRadioButtonIsChecked()) {
+            createPopup(getString(R.string.uncheckedCheckboxMessage));
+        }
+        else {
+            RadioButton oneshotRadioButton = findViewById(R.id.oneshot);
+            String oneShotOrCampaign = oneshotRadioButton.isChecked() ? "Oneshot" : "Campaign";
+            Game newGame = new Game("", findViewById(R.id.create_game_name_field).toString(),
+                                    valueMin, valueMax, difficultySpinner.getSelectedItem().toString(),
+                                    universesSpinner.getSelectedItem().toString(),
+                                    oneShotOrCampaign, findViewById(R.id.numb_session_field).toString(),
+                                    sessionLengthSpinner.getSelectedItem().toString(),
+                                    findViewById(R.id.description_field).toString());
+            Intent intent = new Intent(this, GameList.class);
+            startActivity(intent);
```

Comment:
When calling the new intent, make sure to remove the calling intent from the stack. You may want to take a look at `intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_TASK_ON_HOME)`.

See https://developer.android.com/guide/components/activities/tasks-and-back-stack